warehouse/warehouse.go

```diff
@@ -209,16 +208,20 @@ func (wh *HandleT) syncLiveWarehouseStatus(sourceID string, destinationID string
 		wh.recordDeliveryStatus(uploadID)
 	}
 }
-func getNamespaceFromDestinationConfig(config interface{}, sourceName string, destType string) string {
+
+func getNamespaceFromDestinationConfig(config interface{}, source backendconfig.SourceT, destination backendconfig.DestinationT, destType string) string {
 	configMap := config.(map[string]interface{})
 	var namespace string
 	if configMap["namespace"] != nil {
 		namespace = configMap["namespace"].(string)
 	}
 	if len(strings.TrimSpace(namespace)) > 0 {
-		namespace = misc.TruncateStr(warehouseutils.ToCase(destType, warehouseutils.ToSafeDBString(destType, strcase.ToSnake(namespace))), 127)
+		namespace = warehouseutils.ToCase(destType, warehouseutils.ToSafeNamespace(destType, namespace))
```

Comment:
Minor comment to make it more readable. 

```
	if configMap["namespace"] != nil {
		namespace = configMap["namespace"].(string)
        if len(strings.TrimSpace(namespace)) > 0 {
           return warehouseutils.ToCase(destType, warehouseutils.ToSafeNamespace(destType, namespace))
       }
	}

		namespace, exists := warehouseutils.GetNamespace(source, destination, dbHandle)
		if !exists {
			namespace = warehouseutils.ToCase(destType, warehouseutils.ToSafeNamespace(destType, source.Name))
		}
	return namespace
```

Reply:
Instead of empty string checks, we should start moving towards (value, bool) kind of signatures.

Reply:
changed